internal/controller/factory/configmap.go

```diff
@@ -39,39 +40,16 @@ func CreateOrUpdateClusterStateConfigMap(
 	rclient client.Client,
 ) error {
 	var err error
-	initialCluster := ""
-	clusterService := fmt.Sprintf("%s.%s.svc:2380", GetHeadlessServiceName(cluster), cluster.Namespace)
-	for i := int32(0); i < *cluster.Spec.Replicas; i++ {
-		if i > 0 {
-			initialCluster += ","
-		}
-		podName := fmt.Sprintf("%s-%d", cluster.Name, i)
-		initialCluster += fmt.Sprintf("%s=https://%s.%s",
-			podName, podName, clusterService,
-		)
-	}
-
-	configMap := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: cluster.Namespace,
-			Name:      GetClusterStateConfigMapName(cluster),
-		},
-		Data: map[string]string{
-			"ETCD_INITIAL_CLUSTER_STATE": "new",
-			"ETCD_INITIAL_CLUSTER":       initialCluster,
-			"ETCD_INITIAL_CLUSTER_TOKEN": cluster.Name + "-" + cluster.Namespace,
-		},
+	state := "new"
+	if isEtcdClusterReady(cluster) {
+		state = "existing"
```

Comment:
Do we need deleted log line `log.Debug(ctx, "updating cluster state")` ?

Reply:
IMO, logging changes stored in an in-memory variable is excessive. This is essentially

```go
log.Debug("incrementing a by 1")
a++
```

The correct approach would be

```go
a++
log.Debug("updating a")
client.Update(&a)
```